### ISSUE TITLE
Supermaven is unusable in zed

### ISSUE THREAD

**Issue by kynnyhsap** [zed-industries/zed#13039]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

in vscode supermaven suggest completions at the end of the line and when you hit tab **it replaces the whole line to the end**.

https://github.com/zed-industries/zed/assets/30318777/450b5cfd-dffa-44b5-a750-f163aacc90f9

but in zed suggestions is displayed in the middle of the line where the cursor at and when you hit tab **it only inserts in the middle**

https://github.com/zed-industries/zed/assets/30318777/15813669-5b8d-4626-8f8c-f9592974f3d1

**this ergonomic issue makes supermavan unusable in zed because supermaven suggestions are too fast and my code jumps all the time**

### Environment

Zed: v0.141.0 (Zed Nightly 124878878069a0aa52e97e531fef22fbef43e12f)
OS: macOS 14.1.1
Memory: 32 GiB
Architecture: aarch64

**Comment by xcrap** [comment]
Yes, this completion to the end of the line without replacing or not repeating the end code makes it practically unusable.

**Comment by siddarthvader** [comment]
same issue, it makes the whole editor unresponsive as well, any help or suggestion most welcome.

**Comment by jameelsocorro** [comment]
Same issue. This is the only one I'm waiting to go full zed.

**Comment by xcrap** [comment]
The tab is causing my some problems, I want to add tabs to insert code and it starts completing when I just want to add tab, this really needs some polishing. :)

**Comment by formbook** [comment]
why is this closed?

**Comment by LucasKuhn** [comment]
It even deletes the rest of my existing code when I accept a suggestion 

https://github.com/user-attachments/assets/43c7500f-712b-447d-b88b-219063220a62

**Comment by bujoralexandru** [comment]
+1 My experience is also really bad using supermaven inside Zed. I'm using Zed 0.164.2

https://github.com/user-attachments/assets/6bab6d53-9a06-4325-be0a-ccb09ec85257

**Comment by edhemphill** [comment]
Yeah. I have to say - i like Zed but the UX around auto-completion is pretty horrid. I have resorted to disabling Supermaven all the time. Its constantly is suggesting so often its interrupting my ability to actually edit code.  

It's come to the point that i may need to switch away from this editor. Its just too crappy.  And Supermaven UX in VS Code is just fine. not near as distracting.

This issue should not be closed. The contributors should really check out the issue. It's too fast. It's difficult to tell what its adding. It jumbles up the code.  And the only way to avoid it is to constantly press ESC to clear the suggestion. Move the cursor one char, and boom it will suggest again. Very poor UX on this one y'all.

**Comment by edhemphill** [comment]
I would love contribute - but i am quite overwhelmed. If anyone has the time, here is the logic I was considering:

if the user presses ESC (or whatever action to cancel an auto-complete) a default backoff starts. If the user cancels an auto-complete again, a logarithmic increasing back off happens etc.  that back off should probably reset if the user moves focus to another file or another N or so lines away.  If this type logic was implemented the supermaven issues would be much less a problem.

Another "ask" for a UX I would look for - would be clearer indication of what was auto-complete and what wasn't. I spent sometime looking at Zed docs to see if there was a styling i could find. Anyway, my personal preference would be an 'outline' of the suggestion (b/c the dimmed out stuff is too close to real code and comments) - but this should of course be stylable.

Anyway, i hope these suggestions are at least considered. The would further differentiate Zed from VSC and cursor. The llms will continue to suggest junk, even the best ones, no one is just gonna hit tab all the time and take their stuff.
